src/pk_encryption.rs

```diff
@@ -181,6 +236,35 @@ impl Default for PkDecryption {
     }
 }
 
+impl TryFrom<PkDecryptionPickle> for PkDecryption {
+    type Error = crate::LibolmPickleError;
+
+    fn try_from(pickle: PkDecryptionPickle) -> Result<Self, Self::Error> {
+        let key = Curve25519SecretKey::from_slice(&pickle.private_curve25519_key);
+        let public_key = Curve25519PublicKey::from(&key);
+
+        Ok(Self { key, public_key })
+    }
+}
+
+/// A libolm compatible and picklable form of [`PkDecryption`].
+#[derive(Encode, Decode, Zeroize, ZeroizeOnDrop)]
+struct PkDecryptionPickle {
+    version: u32,
+    public_curve25519_key: [u8; 32],
+    private_curve25519_key: Box<[u8; 32]>,
```

Comment:
This should be annotated as a secret, described here: https://github.com/matrix-org/matrix-pickle/?tab=readme-ov-file#encoding-and-decoding-secrets.